darwin/utils.py

```diff
@@ -210,13 +280,15 @@ def find_files(
     for f in files:
         path = Path(f)
         if path.is_dir():
-            found_files.extend([f for f in path.glob(pattern) if is_extension_allowed("".join(f.suffixes))])
-        elif is_extension_allowed("".join(path.suffixes)):
+            found_files.extend([f for f in path.glob(pattern) if is_extension_allowed(str(path))])
+        elif is_extension_allowed_by_filename(str(path)):
             found_files.append(path)
         else:
             raise UnsupportedFileType(path)
 
-    return [f for f in found_files if f not in map(Path, files_to_exclude)]
+    files_to_exclude_full_paths = [str(Path(f)) for f in files_to_exclude]
```

Comment:
Turned into two lines for readability, because applying the `str` as
```python
[f for f in found file if str(f) not in map(str, map(Path, files_to_exclude))]
```
is not too readable, and `import this` will tell us it should be.

Comparing by string, because comparing `PosixPath`, `WindowsPath`, or generally `PurePath` objects is quite unreliable.  Iteratively comparing `path.parents` and then `path.name` works, and casting to `str` provides `...parents/name` in a single string - which is more efficient to compare.

Comparison of parents and then name is `O(n)=n*(p+1)` - where p is number of parents.  Comparison of string output is `O(n)=1` - at least on the surface - but under the hood, we should at least assume that `pathlib` is relatively optimised.